app/renderer/js/utils/dnd-util.js

```diff
@@ -2,30 +2,168 @@
 
 const ConfigUtil = require(__dirname + '/config-util.js');
 
+const {ipcRenderer} = require('electron');
+
+let count = 0;
+let	dnd;
+let newSettings;
+let time = -1;
+function makeView() {
+	const dndOffTime = {
```

Comment:
This piece of code shouldn't be here as it gets the time even before the user chooses the time delay. It could setoff a some amount of less dnd time.

Reply:
by default, we assume the dnd time to be "selected by user" for which, -1 is taken. In the if statements, the DND time would be added only when the time is positive. So, it would never set-off a less time than the present.